packages/core/core-flows/src/cart/workflows/refresh-cart-items.ts

```diff
@@ -0,0 +1,131 @@
+import { isDefined, PromotionActions } from "@medusajs/framework/utils"
+import {
+  createWorkflow,
+  transform,
+  WorkflowData,
+  WorkflowResponse,
+} from "@medusajs/framework/workflows-sdk"
+import { useRemoteQueryStep } from "../../common/steps/use-remote-query"
+import { refreshCartShippingMethodsStep, updateLineItemsStep } from "../steps"
+import { validateVariantPricesStep } from "../steps/validate-variant-prices"
+import {
+  cartFieldsForRefreshSteps,
+  productVariantsFields,
+} from "../utils/fields"
+import { prepareLineItemData } from "../utils/prepare-line-item-data"
+import { refreshPaymentCollectionForCartWorkflow } from "./refresh-payment-collection"
+import { updateCartPromotionsWorkflow } from "./update-cart-promotions"
+import { updateTaxLinesWorkflow } from "./update-tax-lines"
+
+export const refreshCartItemsWorkflowId = "refresh-cart-items"
+/**
+ * This workflow refreshes a cart's items
+ */
+export const refreshCartItemsWorkflow = createWorkflow(
```

Comment:
discussion: With that new workflow would that mean that in the future we could provide a light add to cart and eventually call the refresh to get all the right taxes and promotions etc?

cc @carlos-r-l-rodrigues

Reply:
Yes, that should be possible. This would require some clever frontend state management for it to work accurately

Reply:
The idea would be to let the option to the front end to for example only show the number of items you have in your cart (which is the case most of the time) and in that case you don't need to always perform the calculation and related update. Then when the customer goes to the cart it could be the place you refresh it. Also, we could have safety system on our side, for example if you complete a cart we check that it is fresh or we refresh it always or something like that. But the main idea would be to remove that bottleneck for most cases but still have the ability to refresh for each add to cart depending on your needs